packages/dashboard/src/components/tasks/task-schedule.tsx

```diff
@@ -0,0 +1,343 @@
+  const { showAlert } = React.useContext(AppControllerContext);
+  const { waypointNames, pickupPoints, dropoffPoints, cleaningZoneNames } =
+    useCreateTaskFormData(rmf);
+  const username = useGetUsername(rmf);
+  const [eventScope, setEventScope] = React.useState<string>(EventScopes.CURRENT);
+  const [refreshTaskAppCount, setRefreshTaskAppCount] = React.useState(0);
+  const exceptDateRef = React.useRef<Date>(new Date());
+  const currentEventIdRef = React.useRef<number>(-1);
+  const [currentScheduleTask, setCurrentScheduledTask] = React.useState<ScheduledTask | undefined>(
+    undefined,
+  );
+  const [calendarEvents, setCalendarEvents] = React.useState<ProcessedEvent[]>([]);
+  const [openDeleteScheduleDialog, setOpenDeleteScheduleDialog] = React.useState(false);
+  const [openCreateTaskForm, setOpenCreateTaskForm] = React.useState(false);
+  const [selectedSchedule, setSelectedSchedule] = React.useState<Schedule>({
+    startOn: new Date(),
+    days: [false, false, false, false, false, false, false],
+    until: undefined,
+  });
+
+  React.useEffect(() => {
+    const sub = AppEvents.refreshTaskApp.subscribe({
+      next: () => {
+        setRefreshTaskAppCount((oldValue) => ++oldValue);
+      },
+    });
+    return () => sub.unsubscribe();
+  }, []);
+
+  const eventsMap = React.useRef<Record<number, ScheduledTask>>({});
+  const getRemoteEvents = React.useCallback<NonNullable<SchedulerProps['getRemoteEvents']>>(
+    async (params) => {
+      if (!rmf) {
+        return;
+      }
+      const tasks = (
+        await rmf.tasksApi.getScheduledTasksScheduledTasksGet(
+          params.end.toISOString(),
+          params.start.toISOString(),
+        )
+      ).data;
+      let counter = 0;
+      const getEventId = () => {
+        return counter++;
+      };
+      eventsMap.current = {};
+      return tasks.flatMap((t: ScheduledTask) =>
+        t.schedules.flatMap<ProcessedEvent>((s: ApiSchedule) => {
+          const events = scheduleToEvents(params.start, params.end, s, t, getEventId, () =>
+            getScheduledTaskTitle(t),
+          );
+          events.forEach((ev) => {
+            eventsMap.current[Number(ev.event_id)] = t;
+          });
+          setCalendarEvents(events);
+          return events;
+        }),
+      );
+    },
+    [rmf],
+  );
+
+  const CustomCalendarEditor = ({ scheduler, value, onChange }: CustomCalendarEditorProps) => {
+    return (
+      <ConfirmationDialog
+        confirmText={'Ok'}
+        cancelText="Cancel"
+        open={true}
+        title={'Edit recurring task'}
+        submitting={undefined}
+        onClose={() => {
+          scheduler.close();
+          setEventScope(EventScopes.CURRENT);
+          AppEvents.refreshTaskApp.next();
+        }}
+        onSubmit={() => {
+          setOpenCreateTaskForm(true);
+          const task = eventsMap.current[Number(currentEventIdRef.current)];
+          if (!task) {
+            throw new Error('No task found');
```

Comment:
We should have consistent error messages similar to `handleSubmitDeleteSchedule`